.travis/before-install-osx.sh

```diff
@@ -2,12 +2,22 @@
 
 export HOMEBREW_NO_ANALYTICS=1
 
-brew install libsndfile || brew install libsndfile || exit 1
+if $UPDATE_HOMEBREW; then
+    #run update first so that possible update errors won't hold up package installation
+    brew update
+else
+    export HOMEBREW_NO_AUTO_UPDATE=1
+fi
+
+brew install libsndfile || exit 1
 brew install portaudio || exit 2
 brew install ccache || exit 3
-brew upgrade qt5 || exit 4
-brew link qt5 --force || exit 5
-brew install fftw --verbose # temp allow
+if  [[ ! -z "$QT_FORMULA" ]]; then
+    brew install $QT_FORMULA --force || exit 4
+elif $UPDATE_HOMEBREW; then
+    brew upgrade qt5 || exit 4
+fi
```

Comment:
after the changes i note above this can become

```bash
if $IS_LEGACY_BUILD; then
  brew install <qt5.9-formula> --force || exit 4
else
  brew upgrade qt5 || exit 4
  brew link --force qt5 || exit 5
fi
```

Reply:
Sounds good.

BTW I was dropping `brew link`. Do you know why is it here? AFAIU it would link qt to a directory in the PATH but I don't think this is needed. We get qt path from homebrew anyway through ``` -DCMAKE_PREFIX_PATH=`brew --prefix qt5` ```. I never needed to link qt to make SC build.

Reply:
oh good point, yeah it can be dropped. thanks!